packages/net-stubbing/test/unit/route-matching-spec.ts

```diff
@@ -41,100 +42,120 @@ describe('intercept-request', function () {
-
-      const matched = _doesRouteMatch({}, req)
-
-      expect(matched).to.be.true
+      }, {})
     })
 
     it('matches on auth matcher', function () {
-      const req = {
+      tryMatch({
         headers: {
           authorization: 'basic Zm9vOmJhcg==',
         },
-        method: 'GET',
         proxiedUrl: 'https://google.com/asdf?1234=a',
-      } as unknown as CypressIncomingRequest
-
-      const matched = _doesRouteMatch({
+      }, {
         auth: {
           username: /^Fo[aob]$/i,
           password: /.*/,
         },
-      }, req)
-
-      expect(matched).to.be.true
+      })
     })
 
     it('doesn\'t match on a partial match', function () {
-      const req = {
+      tryMatch({
         headers: {
           authorization: 'basic Zm9vOmJhcg==',
         },
-        method: 'GET',
         proxiedUrl: 'https://google.com/asdf?1234=a',
-      } as unknown as CypressIncomingRequest
-
-      const matched = _doesRouteMatch({
+      }, {
         auth: {
           username: /^Fo[aob]$/i,
           password: /.*/,
         },
         method: 'POST',
-      }, req)
-
-      expect(matched).to.be.false
+      }, false)
     })
 
     it('handles querystrings as expected', function () {
       const req = {
-        headers: {},
-        method: 'GET',
         proxiedUrl: '/abc?foo=bar&baz=quux',
-      } as unknown as CypressIncomingRequest
+      }
 
-      expect(_doesRouteMatch({
+      tryMatch(req, {
         query: {
           foo: 'b*r',
           baz: /quu[x]/,
         },
-      }, req)).to.be.true
+      })
 
-      expect(_doesRouteMatch({
+      tryMatch(req, {
         path: '/abc?foo=bar&baz=qu*x',
-      }, req)).to.be.true
+      })
 
-      expect(_doesRouteMatch({
+      tryMatch(req, {
         pathname: '/abc',
-      }, req)).to.be.true
+      })
 
-      expect(_doesRouteMatch({
+      tryMatch(req, {
         url: '*',
-      }, req)).to.be.true
+      })
+    })
+
+    context('with matchUrlAgainstPath', function () {
+      it('matches globs against path', function () {
+        tryMatch({
+          proxiedUrl: 'http://foo.com/bar/a1',
+        }, {
+          matchUrlAgainstPath: true,
+          url: '/bar/*',
+        })
+      })
+
+      it('matches nested glob against path', function () {
+        tryMatch({
+          proxiedUrl: 'http://foo.com/bar/a1/foo',
+        }, {
+          matchUrlAgainstPath: true,
+          url: '/bar/*/foo',
+        })
+      })
+
+      it('fails to match with missing queryparams', function () {
+        tryMatch({
+          proxiedUrl: 'http://foo.com/foo/nested?k=v',
+        }, {
+          matchUrlAgainstPath: true,
+          url: '/*/nested',
+        }, false)
+      })
+
+      it('can glob-match against queryparams', function () {
+        tryMatch({
+          proxiedUrl: 'http://foo.com/foo/nested?k=v',
+        }, {
+          matchUrlAgainstPath: true,
+          url: '/*/nested?k=*',
+        })
+      })
```

Comment:
@jennifer-shehane what do you think about this? since it matches against `path`, it matches including querystrings, which means `/*/foo` won't match `/1/foo?a=b` but `/*/foo*` will

Reply:
I feel like this is expected.